contracts/Registry.vy

```diff
@@ -154,13 +174,10 @@ def _get_rates(_pool: address) -> uint256[MAX_COINS]:
             coin: address = self.pool_data[_pool].coins[i]
             if coin == ZERO_ADDRESS:
                 break
-            if coin == self.pool_data[_pool].ul_coins[i]:
+            if coin == self.pool_data[_pool].ul_coins[i] or rate_method_id == 0x00000000:
```

Comment:
Should compare `rate_method_id` first, short circuiting will mean when it evalutes true we avoid a storage load.